base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/views/ScriptGeneratorView.java

```diff
@@ -54,287 +62,320 @@
-
-	/**
-	 * A clear colour for use in other script generator table columns when a row is valid.
-	 */
-	private static final Color clearColor = DISPLAY.getSystemColor(SWT.COLOR_WHITE);
-	
-	/**
-	 * The UI table of script generator contents.
-	 */
-	private ActionsViewTable table;
-	
-	/**
-	 * The drop down box to manipulate which config is being used and load ActionParameters accordingly.
-	 */
-	private ComboViewer configSelector;
-	
-	/**
-	 * The ViewModel the View is updated by.
-	 */
-	private ScriptGeneratorViewModel scriptGeneratorViewModel;
-	
-	/**
-	 * The string to display if there is no configs to select.
-	 */
-	private static final String NO_CONFIGS_MESSAGE = String.format("\u26A0 Warning: Could not load any configs from %s"
-			+ System.getProperty("line.separator")
-			+ "Have they been located in the correct place or is this not your preferred location?", 
-			preferences.scriptGeneratorConfigFolders());
-	
-	/**
-	 * Create a button to manipulate the rows of the script generator table and
-	 *  move them up and down.
-	 * 
-	 * @param parent The composite the button will live in.
-	 * @param icon The icon to display on the button.
-	 * @param direction The direction of the button "up" or "down".
-	 * @return The created button.
-	 */
-	private Button createMoveRowButton(Composite parent, String icon, String direction) {
+    private static final Logger LOGGER = IsisLog.getLogger(ScriptGeneratorView.class);
+
+    private static PreferenceSupplier preferences = new PreferenceSupplier();
+
+    private DataBindingContext bindingContext;
+
+    private static final Display DISPLAY = Display.getDefault();
+
+    /**
+     * A clear colour for use in other script generator table columns when a row is valid.
+     */
+    private static final Color clearColor = DISPLAY.getSystemColor(SWT.COLOR_WHITE);
+
+    /**
+     * The UI table of script generator contents.
+     */
+    private ActionsViewTable table;
+
+    /**
+     * The drop down box to manipulate which config is being used and load ActionParameters accordingly.
+     */
+    private ComboViewer configSelector;
+
+    /**
+     * The ViewModel the View is updated by.
+     */
+    private ScriptGeneratorViewModel scriptGeneratorViewModel;
+
+    /**
+     * The string to display if there is no configs to select.
+     */
+    private static final String NO_CONFIGS_MESSAGE = String.format("\u26A0 Warning: Could not load any configs from %s"
+            + System.getProperty("line.separator")
+            + "Have they been located in the correct place or is this not your preferred location?", 
+            preferences.scriptGeneratorConfigFolders());
+
+    /**
+     * The URL of the Script Generator page on the user manual
+     */
+    private static final URL SCRIPT_GENERATOR_DOCS_URL;
+    static {
+        URL temp = null;
+        try {
+            temp = new URL("http://shadow.nd.rl.ac.uk/ibex_user_manual/Using-the-Script-Generator");
```

Comment:
So with the URLs, we have a situation where the instruments can only access this manual on shadow and shadow is also accessible throughout the ISIS network. However, shadow is inaccessible to those outside of the network. There is a long term goal to distribute the script generator to people outside of the network and possibly to other facilities.

We need to handle these cases. My suggestion would be to be able to set a bunch of URLs as a preference (this can then be set for anyone using the built script generator) and loop through them trying to connect to them.  If we manage to connect to one with a status of 200 or other sucess i.e. 2xx this URL should be used.

Reply:
Please ask me about the use of preferences in IBEX if you need

Reply:
Just to add to James' review: this should be coming from a model, not part of the view code. The suggestion to try connecting to URLs is good but please ensure this is not being done on the UI thread (otherwise we will hang the GUI while waiting for timeouts on unresponsive URLs)

Reply:
The URL is now loaded from the preferences when the singleton is first created